Src/Support/Google.Apis.Core/Http/ConfigurableMessageHandler.cs

```diff
@@ -532,14 +530,23 @@ private void LogHeaders(string initialText, HttpHeaders headers1, HttpHeaders he
                         // Try to handle the abnormal HTTP response with each handler.
                         foreach (var handler in handlers)
                         {
-                            errorHandled |= await handler.HandleResponseAsync(new HandleUnsuccessfulResponseArgs
-                                {
-                                    Request = request,
-                                    Response = response,
-                                    TotalTries = NumTries,
-                                    CurrentFailedTry = NumTries - triesRemaining,
-                                    CancellationToken = cancellationToken
-                                }).ConfigureAwait(false);
+                            try
+                            {
+                                errorHandled |= await handler.HandleResponseAsync(new HandleUnsuccessfulResponseArgs
+                                    {
+                                        Request = request,
+                                        Response = response,
+                                        TotalTries = NumTries,
+                                        CurrentFailedTry = NumTries - triesRemaining,
+                                        CancellationToken = cancellationToken
+                                    }).ConfigureAwait(false);
+                            }
+                            catch
```

Comment:
You could potentially do this in an exception filter, which *might* lead to better exceptions by avoiding the rethrow (even via just throw):

```csharp
catch when (DisposeAndReturnFalse(response)) {}
...
bool DisposeAndReturnFalse(IDisposable disposable)
{
    disposable.Dispose();
    return false;
}
```